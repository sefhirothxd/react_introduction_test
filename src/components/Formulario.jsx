import React from 'react';

const formulario = () => {
  const onSubmit = (e) => {
    e.preventDefault();
    if (
      e.target.elements.email.value.trim() === '' ||
      e.target.elements.password.value.trim() === ''
    ) {
      return;
    }
    console.log(e.target.elements.email.value);
    console.log(e.target.elements.password.value);
  };
  return (
    <form
      onSubmit={onSubmit}
      aria-label="form"
      className="flex flex-col gap-2 items-center"
    >
      <h1 data-testid="test-title">Login</h1>
      <input
        data-testid="test-email"
        type="text"
        name="email"
        aria-label="email"
      />
      <input
        aria-label="password"
        data-testid="test-password"
        type="password"
        role={'textbox'}
        name="password"
      />
      <button type="Submit" className="border border-white">
        enviar
      </button>
    </form>
  );
};

export default formulario;
