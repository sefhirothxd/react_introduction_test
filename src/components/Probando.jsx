import React, { useState } from 'react';

const Probando = () => {
  const [count, setCount] = useState(0);
  return (
    <div>
      <p data-testid="contador">{count}</p>
      <button
        data-testid="botton"
        onClick={() => setCount((count) => count + 2)}
      ></button>
    </div>
  );
};

export default Probando;
