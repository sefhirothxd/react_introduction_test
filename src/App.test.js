import { render, screen, fireEvent } from '@testing-library/react';
import Formulario from './components/Formulario';
import Probando from './components/Probando';

describe('Probando el contador', () => {
  test('probando si el contador inicia en 0', () => {
    render(<Probando />);
    const contador = screen.getByTestId('contador');
    expect(contador.innerHTML).toEqual('0');
  });
  test('probando snapshots', () => {
    const { container } = render(<Probando />);
    expect(container).toMatchSnapshot();
  });
  test('probando si el contador aumenta en 2', () => {
    render(<Probando />);
    const boton = screen.getByTestId('botton');
    fireEvent.click(boton);
    const contador = screen.getByTestId('contador');
    expect(contador.innerHTML).toEqual('2');
  });
});

describe('Haciendo pruebas al componente formulario', () => {
  test('verificando si existe el titulo Login', () => {
    const title = 'Login';
    const { container, getByText, getByTestId } = render(<Formulario />);

    // const h1 = container.querySelector('h1');
    // expect(h1.innerHTML).toEqual(title);
    // toBe se aseguro que sea identico (espacio,etc)
    //ToContent es para verificar si el contenido es el mismo
    //toBeTruthy es para verificar si el elemento existe.
    expect(getByTestId('test-title').innerHTML).toContain(title);
  });
  test('verificando si existe el boton submit', () => {
    const { container, getByTestId } = render(<Formulario />);
    const form = screen.getByRole('form');
    const submit = screen.getByRole('button');
    fireEvent.submit(form);
    expect(submit).toBeTruthy();
  });
  test('verificando si existe el input email y contraseÃ±a', () => {
    const inputEmail = 'probando@gmail.com';
    const inputPassword = '123456';
    const { container, getByTestId } = render(<Formulario />);
    //getByRole es para verificar si existe el elemento con el role especificado.
    const form = screen.getByRole('form');
    const email = screen.getByRole('textbox', { name: 'email' });
    const password = screen.getByRole('textbox', { name: 'password' });
    //fireEvent es para simular el evento click.
    //fireEvent.input es para simular el evento input.
    fireEvent.input(email, { target: { value: inputEmail } });
    fireEvent.input(password, { target: { value: inputPassword } });
    //fireEvent.submit es para simular el evento submit.
    fireEvent.submit(form);
    expect(getByTestId('test-email')).toBeTruthy();
    expect(getByTestId('test-password')).toBeTruthy();
  });
});
