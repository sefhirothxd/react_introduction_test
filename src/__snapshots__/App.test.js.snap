// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el contador probando snapshots 1`] = `
<div>
  <div>
    <p
      data-testid="contador"
    >
      0
    </p>
    <button
      data-testid="botton"
    />
  </div>
</div>
`;
